src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/ExcelAdaptorServiceImpl.java

```diff
@@ -58,9 +66,25 @@
         } else if (sheet.getPhysicalNumberOfRows() < 2) { // check at least 1 row
             throw new ExcelValidationException(HttpStatus.BAD_REQUEST, FILE_NO_DATA_ERROR_MESSAGE);
         }
+        List<String> headers = new LinkedList<>();
+        collectHeaderList(headers, sheet);
+        if (classType.equals(uk.gov.hmcts.reform.cwrdapi.client.domain.CaseWorkerProfile.class)) {
+            validateHeaders(headers);
+        }
         return mapToPojo(sheet, classType);
     }
 
+    private void validateHeaders(List<String> headers) {
+        //below mentioned code can be shortened to headers.containsAll(acceptableHeaders),
+        //but current code is better from debugging standpoint.
+        acceptableHeaders.forEach(acceptableHeader -> {
+            if (!headers.contains(acceptableHeader)) {
+                log.warn(String.format(FILE_MISSING_HEADER_NAME, acceptableHeader));
+                throw new ExcelValidationException(HttpStatus.BAD_REQUEST, FILE_MISSING_HEADERS);
```

Comment:
Can we include the header missing in the exception message so that the consumer knows what is missing ?

Reply:
Updated the code to reveal the missing header.